runtime/tests/js/ipfs_retrieval_tests.js

```diff
@@ -0,0 +1,53 @@
+import { test } from "zinnia:test";
+import { assertEquals, AssertionError } from "zinnia:assert";
+
+const EXPECTED_CAR_BASE64 =
+  "OqJlcm9vdHOB2CpYJQABcBIgO/KicpaH2Kj0sXyJNWLdY4kGpEe2mjY5zovBGRJ+6mpndmVyc2lvbgFrAXASIDvyonKWh9io9LF8iTVi3WOJBqRHtpo2Oc6LwRkSfupqCkUIAhI/TXkgbW9zdCBmYW1vdXMgZHJhd2luZywgYW5kIG9uZSBvZiB0aGUgZmlyc3QgSSBkaWQgZm9yIHRoZSBzaXRlGD8=";
+
+test("can retrieve CID content as a CAR file", async () => {
+  const requestUrl = "ipfs://bafybeib36krhffuh3cupjml4re2wfxldredkir5wti3dttulyemre7xkni";
+
+  const response = await fetch(requestUrl);
+  if (!response.ok) {
+    throw new AssertionError(
+      `Fetch request failed with status code ${response.status}: ${await response.body()}`,
```

Comment:
```suggestion
      `Fetch request failed with status code ${response.status}: ${await response.text()}`,
```

https://developer.mozilla.org/en-US/docs/Web/API/Response/body

Looks like you didn't make this test fail :P (ref to our call about testing best practices)

Reply:
I guess if you had started with the test and then added the fetch implementation, that AssertionError would have been tested. For now I think just fixing the calls is enough. If you wanted you could refactor a test utility like

```js
const assertResponseOk = () => { ... }
```

Reply:
Added a helper and verified that it works as intended:

```
ipfs_retrieval_tests.js
  ✖ can retrieve CID content as a CAR file (5674ms)
Error: AssertionError: Fetch request failed with status code 404: no candidates found

    at assertResponseIsOk (file:///Users/bajtos/src/zinnia/zinnia/runtime/tests/js/ipfs_retrieval_tests.js:48:11)
    at eventLoopTick (ext:core/01_core.js:181:11)
```